starter-code/src/viking.js

```diff
@@ -1,11 +1,165 @@
 // Soldier
-function Soldier () {}
+function Soldier(health, strength) {
+    this.health = health;
+    this.strength = strength;
+}
+
+Soldier.prototype.attack = function () {
+    return this.strength;
+};
+
+Soldier.prototype.receiveDamage = function (damage) {
+    // check for less than zero?
+    this.health -= damage;
+};
 
 // Viking
-function Viking () {}
+function Viking(name, health, strength) {
+    Soldier.call(this, health, strength);
+    this.name = name;
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+
+Viking.prototype.receiveDamage = function (damage) {
+
+    // TA says stupid JavaScript cannot call super method like below, so just duplicate the code
+    // Soldier.prototype.receiveDamage.call(this, damage);
```

Comment:
Why would you do something like that if you are already inheriting vikings from soldier? you already have that method declared as a viking method